docs/proposals/commit-push.md

```diff
@@ -0,0 +1,158 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# commit and push capability proposal
+
+### Motivation
+
+There are some use case to commit a container, and push its resulting image to a registry, for example:
+
+1. For a run-once pod whose result represents an image that should be pushed to a registry for consumption
+by other actors in the system
+
+2. There is someone desire to be able to modify a running container and then commit/push the change so they
+could, for example, make a debug change and then scale up the deployment to test it in a clustered mode,
+without having to start from scratch with building a new image.
+
+The goal of this proposal is to add the commit/push capability, which will allow user commit+push their
```

Comment:
Why should this feature be supported by kubernetes explicitly? Why can't it be built on top of kubernetes?
The fact that kubernetes uses docker or any runtime that provides a commit and push functionality is an implementation detail of the system. A pod must bring in all its dependencies include docker (or rkt) that is needs for CI purposes. If there are issues running certain workloads in pods, like dind, let's fix that instead of adding additional concepts to the API that are potentially not going to work across all deployments.